### PR TITLE
Move more path operations into SPM loader

### PR DESCRIPTION
**Pull request by techhat**
Ping @msteed.

### CODE REVIEW

File: salt/spm/__init__.py [hunk 1]
```diff
@@ -367,7 +367,7 @@ def _remove(self, args):
 
         print('... removing')
 
-        if not os.path.exists(self.opts['spm_db']):
+        if not self.pkgfiles['{0}.path_exists'.format(self.files_prov)](self.opts['spm_db']):
             log.error('No database at {0}, cannot remove {1}'.format(self.opts['spm_db'], package))
```
Comment (msteed) [inline salt/spm/__init__.py]: Did you intend to call `db_exists()` here?

File: salt/spm/__init__.py [hunk 2]
```diff
@@ -381,7 +381,7 @@ def _remove(self, args):
         files = self.pkgdb['{0}.list_files'.format(self.db_prov)](package, self.db_conn)
         dirs = []
         for filerow in files:
-            if os.path.isdir(filerow[0]):
+            if not self.pkgfiles['{0}.path_isdir'.format(self.files_prov)](filerow[0]):
                 dirs.append(filerow[0])
```
Comment (msteed) [inline salt/spm/__init__.py]: Should this read `if self.pkgfiles...`?

File: salt/spm/__init__.py (hunk 1 above)
Comment (techhat) [inline salt/spm/__init__.py]: Yes, thanks. I'll fix it in a moment.

**Comment by techhat** [comment]
@msteed, updated.

**Comment by msteed** [comment]
Looks good to me!

**Pull request merged**